nsxt/resource_nsxt_policy_vtep_ha_host_switch_profile.go

```diff
@@ -0,0 +1,241 @@
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtVtepHAHostSwitchProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtVtepHAHostSwitchProfileCreate,
+		Read:   resourceNsxtVtepHAHostSwitchProfileRead,
+		Update: resourceNsxtVtepHAHostSwitchProfileUpdate,
+		Delete: resourceNsxtVtepHAHostSwitchProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"auto_recovery": {
+				Type:        schema.TypeBool,
+				Description: "Enabled status of autonomous recovery option",
+				Optional:    true,
+				Default:     true,
+			},
+			"auto_recovery_initial_wait": {
+				Type:         schema.TypeInt,
+				Description:  "Start time of autonomous recovery (in seconds)",
+				Optional:     true,
+				Default:      300,
+				ValidateFunc: validation.IntBetween(300, 3600),
+			},
+			"auto_recovery_max_backoff": {
+				Type:         schema.TypeInt,
+				Description:  "Maximum backoff time for autonomous recovery (in seconds)",
+				Optional:     true,
+				Default:      86400,
+				ValidateFunc: validation.IntBetween(3600, 86400),
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Enabled status of VTEP High Availability feature",
+				Optional:    true,
+				Default:     false,
+			},
+			"failover_timeout": {
+				Type:         schema.TypeInt,
+				Description:  "VTEP High Availability failover timeout (in seconds)",
+				Optional:     true,
+				Default:      5,
+				ValidateFunc: validation.IntBetween(0, 60),
+			},
+			"realized_id": {
+				Type:        schema.TypeString,
+				Description: "Computed ID of the realized object",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceNsxtVtepHAHostSwitchProfileExists(id string, connector client.Connector, isGlobalManager bool) (bool, error) {
+	var err error
+	client := infra.NewHostSwitchProfilesClient(connector)
+	structValue, err := client.Get(id)
+	if err == nil {
+		converter := bindings.NewTypeConverter()
+		baseInterface, errs := converter.ConvertToGolang(structValue, model.PolicyBaseHostSwitchProfileBindingType())
+		if errs != nil {
+			return false, errs[0]
+		}
+		base := baseInterface.(model.PolicyBaseHostSwitchProfile)
+
+		resourceType := base.ResourceType
+		if resourceType != infra.HostSwitchProfiles_LIST_HOSTSWITCH_PROFILE_TYPE_POLICYVTEPHAHOSTSWITCHPROFILE {
```

Comment:
I don't think this check is needed, and its actually harmful, since two profiles with same id but different types will not be permitted by NSX. Exists function should return `true` if `id` is present, regardless of profile type.